src/frontend/qt_sdl/DiscordRPC.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+    Copyright 2016-2022 melonDS team
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#include "DiscordRPC.h"
+
+#include "Config.h"
+#include "Platform.h"
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include <discord_rpc.h>
+
+const char* APP_ID          = "1049597419152744458";
```

Comment:
How has this number been derived?

Reply:
That's just the [snowflake](https://discord.com/developers/docs/reference#snowflakes) of the [application](https://discord.com/developers/docs/intro#bots-and-apps) (doesn't go deep into detail about what it is, but an application is just, well, an application on Discord) needed for Rich Presence to function.

It's just needed because RPC requires it to function, and doesn't have much aside from the melonDS icon as part of its "Rich Presence Assets" (you upload an image, give it a name and then can use it as either a small or large image on profiles' activity thing).

Reply:
basically you csn't really use RPC with any text you want, you need to make an application for it in the discord developer page

Reply:
Do you still have questions about this or is this resolved?